### DISCUSSION TITLE
How to show parent list item number in child label?

### DISCUSSION THREAD

**Discussion by edmondop** [brechtm/rinohtype#350] [Q&A]
Hello,
we wanted to have our lists in this format:

```
1.
     1.1
           1.1.1
2.
```

We haven't found a way to do it, the solution would be somehow to override this method here, but we haven't found a way to swap the default https://github.com/brechtm/rinohtype/blob/master/src/rinoh/structure.py#L176 LabelItemStyle with a custom class

**Comment by brechtm** [comment]
This type of numbering was already supported for headings and paragraphs. I have added this for lists as well in e0672f23fab99518189705100f1e3b5bece97441.

Install the development version of rinohtype:
```sh
pip install https://github.com/brechtm/rinohtype/archive/refs/heads/master.zip
```

And override the _enumerated list item label_ style definition in your style sheet:
```ini
[enumerated list item label]
number_separator = '.'
```

Please let me know if that works for you.

**Reply by edmondop** [reply]
Thank you, it worked like a charm. Is there a way to have that kind of extensibility that would have allowed us to do it without your help?

**Reply by brechtm** [reply]
That really depends on exactly what you want to achieve. It's impossible to describe what this encompasses exactly, but _most_ styling should be possibly by means of a custom style sheet, template configuration and document template. reStructuredText offers ways of setting a class for document elements, which can be used in styling. This is not properly documented yet, but you see an example here: [Page Break after every section](https://stackoverflow.com/a/65952901/438249).

For more specialized styling, you can define custom reStructuredText roles and directives. You can find some more information on this here:
- StackOverflow: [inline code background color](https://stackoverflow.com/questions/65825828/sphinx-rinohtype-inline-code-background-color)
- #201 (and linked issues)

In summary: there are quite some options, but the lack of documentation makes it more difficult than it should be.
